ambry-clustermap/src/main/java/com.github.ambry.clustermap/AmbryDatanode.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.clustermap;
+
+import com.github.ambry.config.ClusterMapConfig;
+import com.github.ambry.network.Port;
+import com.github.ambry.network.PortType;
+import com.github.ambry.utils.Utils;
+import java.util.List;
+import java.util.TreeSet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.github.ambry.clustermap.ClusterMapUtils.*;
+
+
+/**
+ * {@link DataNodeId} implementation to use within dynamic cluster managers.
+ */
+class AmbryDataNode extends DataNodeId implements Resource {
```

Comment:
is there a reason `DataNode` cannot be folded into this i.e. get the static cluster manager to use the same class? Would prefer that since it is gets rid of possibly repeated code.